src/main/java/com/atherys/towns/facade/TownFacade.java

```diff
@@ -535,11 +535,20 @@ public Text renderTowns(Collection<Town> towns) {
         return townsText.build();
     }
 
-    public void onPlayerDamage(DamageEntityEvent event, Player player) {
-        plotService.getPlotByLocation(player.getLocation()).ifPresent(plot -> {
-            Town townAtPlot = plot.getTown();
-            event.setCancelled(!townAtPlot.isPvpEnabled());
-        });
+    public void onPlayerDamage(DamageEntityEvent event, Player attacker, Player target) {
+        Optional<Plot> attackerPlot = plotService.getPlotByLocation(attacker.getLocation());
+        Optional<Plot> targetPlot = plotService.getPlotByLocation(target.getLocation());
+        //If the attacker and the target are not in plots, then return.
+        if (!(attackerPlot.isPresent() || targetPlot.isPresent())) {
```

Comment:
If(!attackersPlot.isPresent() && !targetPlot.isPresent()) does the same but might be more readable

Reply:
This has been changed.